### PR TITLE
Metrics Span Multiplier

### LINKED ISSUE #2166: Dynamic Sampling
**Is your feature request related to a problem? Please describe.**
In one of our services we want to sample every route, user, app version by another ratio, the problem is generated metrics won't reflect the reality.

**Describe the solution you'd like**
using float64 attribute named `X-SampleRatio` we can multiply all the metrics by `1.0/ratio` 

**Additional context**
It's just like what HoneyComb does in https://github.com/honeycombio/honeycomb-opentelemetry-go/blob/main/deterministic_sampler.go and https://docs.honeycomb.io/manage-data-volume/sampling/

If it's okay i would like to implement it.

### PR DESCRIPTION
**Pull request by altanozlu**
**What this PR does**:
Adds span multiplier feature by using "X-SampleRatio" attribute.

**Which issue(s) this PR fixes**:
Fixes #2166

**Checklist**
- [x] Tests updated
- [x] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: modules/generator/processor/servicegraphs/config.go [hunk 1]
```diff
@@ -27,6 +27,9 @@ type Config struct {
 	// If client and server spans have the same attribute, behaviour is undetermined
 	// (either value could get used)
 	Dimensions []string `yaml:"dimensions"`
+
+	// If enabled X-SampleRatio attribute will be used for metric calculation
+	EnableSpanMultiplier bool `yaml:"enable_span_multiplier"`
```
Comment (mapno) [inline modules/generator/processor/servicegraphs/config.go:32]: If there isn't a semantic convention for this attribute, we could use a string, that if defined, will be used as key to get the information.

**Review by mapno — COMMENTED** [review]
I like the changes. This is something we've wanted to do for a long time. Thanks for starting the work.

File: modules/generator/processor/servicegraphs/config.go (hunk 1 above)
Comment (altanozlu) [reply modules/generator/processor/servicegraphs/config.go:32]: Yea we can have that

**Comment by altanozlu** [comment]
Hi @mapno, do you have any idea why "Test Packages" test failed and can we rerun it ?

**Review by mapno — APPROVED** [review]
LGTM

**Pull request merged**